### PR TITLE
Speculative fix for getCurrentTabURL

### LINKED ISSUE #1214: Extension Error: You must provide a url to the driver
**Lighthouse Version**: 1.3.0
**Chrome Version**: 55.0.2883.95
**Error Message**: You must provide a url to the driver
**Stack Trace**:
 ```Error: You must provide a url to the driver
    at Function.run (chrome-extension://blipmdconlkpinefehnmjammfjpmpbjk/scripts/lighthouse-background.js:11221:29)
    at window.runLighthouseForConnection (chrome-extension://blipmdconlkpinefehnmjammfjpmpbjk/scripts/lighthouse-background.js:12188:17)
    at connection.getCurrentTabURL.then.url (chrome-extension://blipmdconlkpinefehnmjammfjpmpbjk/scripts/lighthouse-background.js:12201:25)```

### PR DESCRIPTION
**Pull request by paulirish**
I am hoping this will fix #1214, but since the repro is particularly tricky, we won't know until later.

I believe is ultimately a chrome bug that's behavior is here: [546696 - chrome.windows.getLastFocused returns the last opened window, not the last focused, when an extension popup is visible - chromium](https://bugs.chromium.org/p/chromium/issues/detail?id=546696#c4) but essentially its interplay between extensions with `onFocusChanged` `getLastFocused` and here, `chrome.tabs.query({lastFocusedWindow: true})`..

Of note, previous to the currentTab change in #680 I don't believe we had this bug, but we did have trouble when debugging the extension background page with a devtools window. I'm OK with regressing our developer ergonomics to get this user bug squashed.

http://stackoverflow.com/a/28794338 (by Rob W) describes the best solution to be:
```js
chrome.tabs.query({
    active: true,
    currentWindow: true    
}
```

The best practices around getting current tab have changed throughout the past 7 years, and there appear to be some still open bugs, so this is what we're running up against.

This PR adds extra error logging around all this so we'll have more information if the problem persists.

### CODE REVIEW

File: lighthouse-core/gather/connections/extension.js [hunk 1]
```diff
@@ -157,7 +159,12 @@ class ExtensionConnection extends Connection {
    * Used by lighthouse-background to kick off the run on the current page
    */
   getCurrentTabURL() {
-    return this._queryCurrentTab().then(tab => tab.url);
+    return this._queryCurrentTab().then(tab => {
+      if (!tab.url) {
+        log.error('ExtensionConnection', 'getCurrentTabURL returned empty string', tab);
```
Comment (ebidel) [inline lighthouse-core/gather/connections/extension.js]: Is logging the tab obj going to be useful?

Comment (paulirish) [inline lighthouse-core/gather/connections/extension.js]: Indeterminable

**Comment by paulirish** [comment]
@mblayman re #1745. would you be willing to see if this PR fixes things for you?

I've uploaded a build of the extension and attached to this comment. You can load it in developer mode (and disable the other lighthouse) to give it a try.

➡️  [lighthouse-1.5.99999.zip](https://github.com/GoogleChrome/lighthouse/files/791857/lighthouse-1.5.99999.zip)

**Comment by mblayman** [comment]
@paulirish I just tried the extension you supplied and I still have the same behavior for `https://storybird.com`. It hits the same timeout that I reported in #1745. 🤔

**Comment by paulirish** [comment]
okay thanks for giving it a go!

**Pull request merged**